### ISSUE TITLE
SSIM are negative

### ISSUE THREAD

**Issue by icey-zhang** [Po-Hsun-Su/pytorch-ssim#31]
Hello, what's wrong with code that comes out with negative values？

**Comment by piotrkawa** [comment]
Hello!

I assume You refer to the class-based SSIM (`pytorch_ssim.SSIM()`), is that correct?
If so:
it seems for me that for the same inputs (and `window_size`) `pytorch_ssim.ssim` and `pytorch_ssim.SSIM()` calculate the very same value. The thing is that class-based SSIM returns the negation of this value - which makes sense, as it's useful in the training process.

Best regards
Piotr

**Comment by nickdalfarra** [comment]
I'm not 100% sure, but there seem to be different conventions on the range of SSIM implementations. Some output to [0,1], others to [-1,1]. I don't have an example of two images producing a perfectly bad SSIM of -1, but you could always just linearly transform the SSIM output via (SSIM + 1)/2 to make the range [0,1].

**Comment by TalTaiber** [comment]
> Hello, what's wrong with code that comes out with negative values？

Just a guess, since the author did not repond:
Higher SSIM corresponds to greater similarity between the two images. However, notice that in the author's explanation we have:
`ssim_loss = pytorch_ssim.SSIM()`
Since the optimizer tries to minimize the loss, we have to define the SSIM loss as negative SSIM, so that when the loss is low the SSIM will be high.
